### PR TITLE
make_deb automated to check underlying arch and build dependency

### PR DESCRIPTION
**Pull request by Singhal-Vaibhav**
**Changes**:
1. creation of debian packages checks for underlying architecture.
2. Removed DEBAIN files of x86 as this plugin is supported only on raspberry pi
3. If package is not built before creation of debian, the project is build first
4. removed preinst for debian packages, not required.

### CODE REVIEW

File: make_deb [hunk 1]
```diff
@@ -108,12 +97,37 @@ if [ ! -L "${BUILD_ROOT}" -a ! -d "${BUILD_ROOT}" ]; then
...
+    fi
+
+    # Check/set LD_LIBRARY_PATH
+    libPathSet=0
+    libdir=${FOGLAMP_ROOT}/lib; [ -d ${libdir} ] && LD_LIBRARY_PATH=$(echo $LD_LIBRARY_PATH | sed "s|${libdir}||g") && export LD_LIBRARY_PATH=${libdir}:${LD_LIBRARY_PATH} && libPathSet=1
+    libdir=${FOGLAMP_ROOT}/cmake_build/C/lib; [ -d ${libdir} ] && LD_LIBRARY_PATH=$(echo $LD_LIBRARY_PATH | sed "s|${libdir}||g") && export LD_LIBRARY_PATH=${libdir}:${LD_LIBRARY_PATH} && libPathSet=1
+    [ "$libPathSet" -eq "0" ] && echo "Unable to set/update LD_LIBRARY_PATH to include path of Foglamp shared libraries: check whether ${FOGLAMP_ROOT}/lib or ${FOGLAMP_ROOT}/cmake_build/C/lib exists" && exit 1
+
+    # Build wiringpi first
+    cd /tmp; git clone git://git.drogon.net/wiringPi; cd wiringPi; ./build; cd ${GIT_ROOT}
```
Comment (ashish-jabble) [inline make_deb]: Do `clone` only if it does not exist or remove `wiringPi` dir before clone

**Review by pintomax — DISMISSED** [review]

File: packages/Debian/armhf/DEBIAN/preinst [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+is_foglamp_running () {
+    set +e
+    foglamp_script=$(get_foglamp_script)
+    foglamp_status_output=$($foglamp_script status 2>&1 | grep 'FogLAMP Uptime')
+    rc=$((!$?))
+    echo $rc
+    set -e
+}
```
Comment (ashish-jabble) [inline packages/Debian/armhf/DEBIAN/preinst]: We don't need to check foglamp running status. See https://scaledb.atlassian.net/browse/FOGL-1817

File: packages/Debian/armhf/DEBIAN/preinst [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+is_plugin_installed () {
+    set +e
+    current_files_all=$(dpkg -L ${PKG_NAME} | grep 'foglamp/plugins/south/dht/lib*')
+    rc=$((!$?))
+    echo $rc
+    set -e
+}
```
Comment (ashish-jabble) [inline packages/Debian/armhf/DEBIAN/preinst]: Even I am not sure about this. I believe same comments apply as above

File: make_deb [hunk 4]
```diff
@@ -21,68 +21,57 @@
...
+         exit 1
+        ;;
+  esac
+done
+
 
 # If the architecture has not been defined, then the script is complete
-if [[ "$architecture" == "none" ]]; then
-  exit 0
+if [ "$(dpkg --print-architecture)" != "armhf" ]; then
```
Comment (ashish-jabble) [inline make_deb:68]: I would suggest `if [ "$(dpkg --print-architecture)" != "arm*" ];` then to keep it generic for ARM

See conversation on [FOGL-1470](https://scaledb.atlassian.net/browse/FOGL-1470?focusedCommentId=39432&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-39432)

**Review by ashish-jabble — DISMISSED** [review]
And `./makedeb` execution is failed.
```
-- Using FOGLAMP_ROOT for includes
   +/usr/local/foglamp//C/thirdparty/rapidjson/include
-- Using FOGLAMP_ROOT for libs 
   +/usr/local/foglamp//cmake_build/C/lib
CMake Error at FindFogLAMP.cmake:105 (message):
  FogLAMP has not been built yet in /usr/local/foglamp/ Compile it first.
Call Stack (most recent call first):
  CMakeLists.txt:39 (find_package)


CMake Error at CMakeLists.txt:47 (message):
  FogLAMP plugin 'dht11' build error
```
Seems like `sudo make install and $FOGLAMP_ROOT set` prerequisite are not enough to make C-debians. It requires headers file too to proper compile plugin (So `make` is required and set FOGLAMP_ROOT this path)
Also, fix references of `x86_64` text in ReadMe(makedeb O/P)

File: make_deb (hunk 4 above)
Comment (Singhal-Vaibhav) [reply make_deb:68]: We cant keep it generic for the below mentioned reasons:
1. Its not tested on other arm architectures - armel and arm64
2. Even if we test that for 1, we need to provide a Control file for that particular architecture. I am not sure if we want to do that currently.  Note packages/Debian/armhf/DEBIAN/control:Architecture

**Comment by Singhal-Vaibhav** [comment]
README fixed and All requested changes done other than making arm generic instead of armhf

**Review by MarkRiddoch — APPROVED** [review]

**Pull request merged**